### PR TITLE
Allow metrics exceptions skipping during formatting

### PR DESCRIPTION
**Pull request by tarampampam**
| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes

### Added

- Interface `ShouldBeSkippedmetricExceptionInterface`
- Exception `ShouldBeSkippedmetricException`
- Possibility to skip metrics if during construction or formatting was thrown exception that implements `ShouldBeSkippedmetricExceptionInterface`. In this case exception would be reported (not thrown out) by `Illuminate\Contracts\Debug\ExceptionHandler` and metric would be skipped from output.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in [CHANGELOG.md](https://github.com/avto-dev/app-metrics-laravel/blob/master/CHANGELOG.md) file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog][keepachangelog] and this project adheres to [Semantic Versioning][semver].
 
+## v2.2.0
+
+### Added
+
+- Interface `ShouldBeSkippedmetricExceptionInterface`
+- Exception `ShouldBeSkippedmetricException`
+- Possibility to skip metrics if during construction or formatting was thrown exception that implements `ShouldBeSkippedmetricExceptionInterface`. In this case exception would be reported (not thrown out) by `Illuminate\Contracts\Debug\ExceptionHandler` and metric would be skipped from formatters output.
```
Comment (tarampampam) [inline CHANGELOG.md:13]: `### Changed`

File: src/Exceptions/ShouldBeSkippedMetricException.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AvtoDev\AppMetrics\Exceptions;
+
+use RuntimeException;
+
+class ShouldBeSkippedMetricException extends RuntimeException implements ShouldBeSkippedMetricExceptionInterface
```
Comment (tarampampam) [inline src/Exceptions/ShouldBeSkippedMetricException.php]: del

File: src/Formatters/JsonFormatter.php [hunk 3]
```diff
@@ -34,14 +51,22 @@ public function format(iterable $metrics, int $options = \JSON_UNESCAPED_UNICODE
...
+                if ($metric instanceof MetricsGroupInterface) {
+                    foreach ($metric->metrics() as $collection_item) {
+                        if ($collection_item instanceof MetricInterface) {
+                            $result[] = (object) $this->metricToArray($collection_item);
+                        }
                     }
+                } elseif ($metric instanceof MetricInterface) {
+                    $result[] = (object) $this->metricToArray($metric);
+                }
+            } catch (ShouldBeSkippedMetricExceptionInterface $e) {
```
Comment (tarampampam) [inline src/Formatters/JsonFormatter.php:67]: Not DRY

File: src/ServiceProvider.php [hunk 4]
```diff
@@ -70,10 +71,12 @@ protected function registerMetricsManager(): void
             static function (Container $container): MetricsManagerInterface {
                 /** @var ConfigRepository $config */
                 $config = $container->make(ConfigRepository::class);
+                $exception_handler = $container->make(ExceptionHandler::class);
```
Comment (tarampampam) [inline src/ServiceProvider.php:75]: Add `/** @var ExceptionHandler $exception_handler */`

**Review by tarampampam — COMMENTED** [review]
After review:

File: src/Formatters/JsonFormatter.php (hunk 3 above)
Comment (Reallife) [reply src/Formatters/JsonFormatter.php:67]: Moved to trait

File: src/ServiceProvider.php (hunk 4 above)
Comment (Reallife) [reply src/ServiceProvider.php:75]: Added

File: CHANGELOG.md (hunk 1 above)
Comment (Reallife) [reply CHANGELOG.md:13]: Added `Changed` section

File: src/Exceptions/ShouldBeSkippedMetricException.php (hunk 2 above)
Comment (Reallife) [reply src/Exceptions/ShouldBeSkippedMetricException.php]: Moved to stubs. I need it in tests.

**Review by tarampampam — COMMENTED** [review]
Looks great!

**Pull request merged**